### ISSUE TITLE
DPDK submodule doesn't build under FreeBSD 14

### ISSUE THREAD

**Issue by mikeBashStuff** [spdk/spdk#3363]
```
SPDK: latest
```
Due to https://github.com/freebsd/freebsd-src/commit/9e4e7bcaef3fcaab66cdecd08b1664ac92cc9a6b
Fixed in DPDK here:   https://github.com/DPDK/dpdk/commit/5eedf66ac0e
cherry-pick: https://review.spdk.io/gerrit/c/spdk/dpdk/+/22999 (applied cleanly).

When the cherry-pick lands, we should point our submodule at it.

Failure looks similar to:
```
FAILED: kernel/freebsd/nic_uio.ko
/usr/bin/make -f ../kernel/freebsd/BSDmakefile.meson KMOD_OBJDIR=kernel/freebsd KMOD_SRC=../kernel/freebsd/nic_uio/nic_uio.c KMOD=nic_uio 'KMOD_CFLAGS=-I/root/spdk/dpdk/build-tmp -I/root/spdk/dpdk/config -include rte_config.h' CC=clang
clang  -O2 -pipe -include rte_config.h  -fno-strict-aliasing -Werror -D_KERNEL -DKLD_MODULE -nostdinc  -I/root/spdk/dpdk/build-tmp -I/root/spdk/dpdk/config -include /root/spdk/dpdk/build-tmp/kernel/freebsd/opt_global.h -I. -I/usr/src/sys -I/usr/src/sys/contrib/ck/include -fno-common  -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fdebug-prefix-map=./machine=/usr/src/sys/amd64/include -fdebug-prefix-map=./x86=/usr/src/sys/x86/include -fdebug-prefix-map=./i386=/usr/src/sys/i386/include     -MD  -MF.depend.nic_uio.o -MTnic_uio.o -mcmodel=kernel -mno-red-zone -mno-mmx -mno-sse -msoft-float  -fno-asynchronous-unwind-tables -ffreestanding -fwrapv -fstack-protector -Wall -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith -Wcast-qual -Wundef -Wno-pointer-sign -D__printf__=__freebsd_kprintf__ -Wmissing-include-dirs -fdiagnostics-show-option -Wno-unknown-pragmas -Wno-error=tautological-compare -Wno-error=empty-body -Wno-error=parentheses-equality -Wno-error=unused-function -Wno-error=pointer-sign -Wno-error=shift-negative-value -Wno-address-of-packed-member -Wno-format-zero-length   -mno-aes -mno-avx  -std=gnu99 -c /root/spdk/dpdk/kernel/freebsd/nic_uio/nic_uio.c -o nic_uio.o
/root/spdk/dpdk/kernel/freebsd/nic_uio/nic_uio.c:84:81: error: too many arguments provided to function-like macro invocation
DRIVER_MODULE(nic_uio, pci, nic_uio_driver, nic_uio_devclass, nic_uio_modevent, 0);
                                                                                ^
/usr/src/sys/sys/bus.h:832:9: note: macro 'DRIVER_MODULE' defined here
#define DRIVER_MODULE(name, busname, driver, evh, arg)                  \
        ^
/root/spdk/dpdk/kernel/freebsd/nic_uio/nic_uio.c:84:1: error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Werror,-Wimplicit-int]
DRIVER_MODULE(nic_uio, pci, nic_uio_driver, nic_uio_devclass, nic_uio_modevent, 0);
^
int
2 errors generated.
*** Error code 1

```

**Comment by mikeBashStuff** [comment]
The https://review.spdk.io/gerrit/c/spdk/dpdk/+/22999 got merged to `spdk-23.11`,  this  https://review.spdk.io/gerrit/c/spdk/spdk/+/23003  bumps the submodule to point at that particular commit.

**Comment by mikeBashStuff** [comment]
All patches merged, closing.
